Subject: Linux-RT 2.6.24-rc5-rt1
From: http://www.kernel.org/pub/linux/kernel/projects/rt/
Acked-by: Sven-Thorsten Dietrich <sdietrich@suse.de>
From sshtylyov@ru.mvista.com Mon Sep 18 18:12:44 2006
Return-Path: <sshtylyov@ru.mvista.com>
Received: from imap.sh.mvista.com (unknown [63.81.120.155]) by mail.tglx.de
	(Postfix) with ESMTP id 0B6FF65C003 for <tglx@linutronix.de>; Mon, 18 Sep
	2006 18:12:44 +0200 (CEST)
Received: from [192.168.1.248] (unknown [10.150.0.9]) by imap.sh.mvista.com
	(Postfix) with ESMTP id 78E423ECB; Mon, 18 Sep 2006 09:12:40 -0700 (PDT)
Message-ID: <450EC5FF.1070103@ru.mvista.com>
Date: Mon, 18 Sep 2006 20:14:55 +0400
From: Sergei Shtylyov <sshtylyov@ru.mvista.com>
Organization: MontaVista Software Inc.
User-Agent: Mozilla/5.0 (X11; U; Linux i686; rv:1.7.2) Gecko/20040803
X-Accept-Language: ru, en-us, en-gb
MIME-Version: 1.0
To: Thomas Gleixner <tglx@linutronix.de>, John Stultz <johnstul@us.ibm.com>
Cc: "Mark A. Greer" <mgreer@mvista.com>, Vitaly Bordug <vbordug@ru.mvista.com>, Vitaly Wool <vwool@ru.mvista.com>
Subject: [PATCH] PPC timebase clocksource is continuous
Content-Type: multipart/mixed; boundary="------------090305050803080806050803"
X-Spam-Checker-Version: SpamAssassin 2.64 (2004-01-11) on debian
X-Spam-Level: 
X-Spam-Status: No, hits=-2.7 required=5.0 tests=BAYES_00,DOMAIN_BODY 
	autolearn=no version=2.64
X-Evolution-Source: imap://tglx%40linutronix.de@imap.tglx.de/

This is a multi-part message in MIME format.
--------------090305050803080806050803
Content-Type: text/plain; charset=us-ascii; format=flowed
Content-Transfer-Encoding: 8bit

Mark the timebase clocksource as continuous since it obviously is.
Fix some whitespace and letter case while at it. :-)

Signed-off-by: Sergei Shtylyov <sshtylyov@ru.mvista.com>


--------------090305050803080806050803
Content-Type: text/plain; name="ppc-timebase-is-continuous.patch"
Content-Disposition: inline; filename="ppc-timebase-is-continuous.patch"
Content-Transfer-Encoding: 8bit

 arch/powerpc/kernel/time.c |   20 ++++++++++----------
 1 file changed, 10 insertions(+), 10 deletions(-)

Index: linux-2.6.23.1-rt11/arch/powerpc/kernel/time.c
===================================================================
--- linux-2.6.23.1-rt11.orig/arch/powerpc/kernel/time.c
+++ linux-2.6.23.1-rt11/arch/powerpc/kernel/time.c
@@ -529,7 +529,7 @@ void timer_interrupt(struct pt_regs * re
 		if (per_cpu(last_jiffy, cpu) >= tb_next_jiffy) {
 			tb_last_jiffy = tb_next_jiffy;
 			do_timer(1);
-			timer_recalc_offset(tb_last_jiffy);
+			/*timer_recalc_offset(tb_last_jiffy);*/
 			timer_check_rtc();
 		}
 		write_sequnlock(&xtime_lock);
@@ -926,22 +926,23 @@ void div128_by_32(u64 dividend_high, u64
 
 }
 
-
-/* powerpc clocksource code */
+/* PowerPC clocksource code */
 
 #include <linux/clocksource.h>
+
 static cycle_t timebase_read(void)
 {
 	return (cycle_t)get_tb();
 }
 
 struct clocksource clocksource_timebase = {
-	.name = "timebase",
-	.rating = 200,
-	.read = timebase_read,
-	.mask = (cycle_t)-1,
-	.mult = 0,
-	.shift = 22,
+	.name		= "timebase",
+	.rating		= 200,
+	.read		= timebase_read,
+	.mask		= (cycle_t)-1,
+	.mult		= 0,
+	.shift		= 22,
+	.is_continuous	= 1,
 };
 
 
@@ -957,4 +958,3 @@ static int __init init_timebase_clocksou
 }
 
 module_init(init_timebase_clocksource);
-
